### PR TITLE
New migrations doc

### LINKED ISSUE #64: Write guidance for using Entity Framework migrations with Aspire solutions
### Type of issue

Missing information

### Description

my AppHost program.cs:

```
var builder = DistributedApplication.CreateBuilder(args);

var sql = builder.AddSqlServerContainer("sql",password:"P4s$wordBuT$tr0ng",port:14420).AddDatabase("sqldata");

builder.AddProject<Projects.adminendpoint>("admin").WithReference(sql);

builder.Build().Run();
```

Then i want to run it but first make database from my dbcontext that used in admin endpoint.
admin endpoint program.cs is:
```
builder.AddSqlServerDbContext<DataBaseContext>("sqldata");
builder.Services.AddScoped<IDataBaseContext, DataBaseContext>();
```
and my dbcontext is :
```
public class DataBaseContext : DbContext, IDataBaseContext
{
    public DataBaseContext(DbContextOptions dbContextOptions):base(dbContextOptions) { }
    public DbSet<User> Users { get; set; }
    public DbSet<Role> Roles { get; set; }
    public DbSet<UserInRole> UserInRoles { get; set; }
}
```

So how should i use Add-Migration and Update-Database?

### Page URL

https://learn.microsoft.com/en-us/dotnet/aspire/database/sql-server-entity-framework-component?tabs=dotnet-cli

### Content source URL

https://github.com/dotnet/docs-aspire/blob/main/docs/database/sql-server-entity-framework-component.md

### Document Version Independent Id

2cd52f56-0ee5-8f04-c2b5-fe52c7f8109f

### Article author

IEvangelist

---
[Associated WorkItem - 186558](https://dev.azure.com/msft-skilling/Content/_workitems/edit/186558)

### LINKED ISSUE #488: Documentation and samples of Aspire + EF Core tooling
Follow on from https://github.com/dotnet/aspire/issues/398

There is an EF migrations playground app that can be used as a starting place for samples.

- [x] Sample - done here https://learn.microsoft.com/en-us/samples/dotnet/aspire-samples/aspire-efcore-migrations/
- [ ] Documentation

---
[Associated WorkItem - 232757](https://dev.azure.com/msft-skilling/Content/_workitems/edit/232757)

### PR DESCRIPTION
**Pull request by CamSoper**
## Summary

New EF Core Migrations doc

Fixes #488 
Fixes #64

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/database/ef-core-migrations.md](https://github.com/dotnet/docs-aspire/blob/6c7502000469ddbded9ab8f3972810b63136f5de/docs/database/ef-core-migrations.md) | [Apply Entity Framework Core migrations in .NET Aspire](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/ef-core-migrations?branch=pr-en-us-838) |

### CODE REVIEW

**Comment by IEvangelist** [comment]
I love this! It's very well written and great content. I have one suggestion, I think we should now add a new TOC entry with both the "Seed" and "Migration" content under it, perhaps named "Database initialization and maintenance"? Thoughts?

**Comment by CamSoper** [comment]
@IEvangelist Makes sense to me. Leave the hub page the way it is, though?

**Comment by IEvangelist** [comment]
> @IEvangelist Makes sense to me. Leave the hub page the way it is, though?

Yes, I'd leave that as is.

**Review by IEvangelist — APPROVED** [review]

File: docs/database/ef-core-migrations.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+]
+```
+
+## Create migrations
+
+Start by creating some migrations to apply.
+
+1. Open a terminal (<kbd>Ctrl</kbd>+<kbd>\`</kbd> in Visual Studio).
+1. Set *SupportTicketApi\SupportTicketApi.Api* as the current directory.
+1. Use the [`dotnet ef` command-line tool](https://learn.microsoft.com/ef/core/managing-schemas/migrations/#install-the-tools) to create a new migration to capture the initial state of the database schema:
```
Comment (IEvangelist) [inline docs/database/ef-core-migrations.md:49]:
```suggestion
1. Use the [`dotnet ef` command-line tool](/ef/core/managing-schemas/migrations/#install-the-tools) to create a new migration to capture the initial state of the database schema:
```

**Pull request merged**